src/units/abilities.cpp

```diff
@@ -839,19 +839,18 @@ std::vector<std::pair<t_string, t_string>> attack_type::special_tooltips(
  * @param[in] sp reference to ability to check
  * @param[in,out] checking_name the reference for checking if @name already added
  */
-static void add_name(std::string& weapon_abilities, bool active, const config::any_child sp, std::set<std::string>& checking_name, bool affect_adjacent)
+static void add_name(std::string& weapon_abilities, bool active, const config::any_child sp, std::set<std::string>& checking_name, bool already_used, const std::string& from_str)
```

Comment:
This takes `already_used` by value rather than by reference; the `already_used = true` later in this function only sets a local variable that isn't read afterwards.